src/CMakeLists.txt

```
@@ -58,7 +58,11 @@ set_target_properties(popsift PROPERTIES DEBUG_POSTFIX "d")
 
 # cannot use PRIVATE here as there is a bug in FindCUDA and CUDA_ADD_LIBRARY
 # https://gitlab.kitware.com/cmake/cmake/issues/16097
-target_link_libraries(popsift ${Boost_LIBRARIES} ${CUDA_CUDADEVRT_LIBRARY} ${CUDA_CUBLAS_LIBRARIES})
+target_link_libraries(popsift ${CUDA_CUDADEVRT_LIBRARY} ${CUDA_CUBLAS_LIBRARIES})
+
+# Link to threads because we're using C++11 std::thread.
+find_package(Threads REQUIRED)
+target_link_libraries(popsift ${CMAKE_THREAD_LIBS_INIT})
```

```suggestion
target_link_libraries(popsift Threads::Threads)
```
Also you can move `find_package(Threads REQUIRED)` up and use a single instruction
```cmake
target_link_libraries(popsift ${CUDA_CUDADEVRT_LIBRARY} ${CUDA_CUBLAS_LIBRARIES} Threads::Threads)
```

---

Switched target name to `Threads::Threads`, and moved `find_package` to the root CMakeLists.txt.